model/histogram/convert_test.go

```diff
@@ -0,0 +1,200 @@
+}
+
+type ExpectedHistogram struct {
+	buckets []BucketExpectation
+	count   float64
+	sum     float64
+}
+
+func TestConvertNHCBToClassicHistogram(t *testing.T) {
+	tests := []struct {
+		name      string
+		nhcb      any
+		labels    labels.Labels
+		expectErr bool
+		expected  ExpectedHistogram
+	}{
+		{
+			name: "Valid Histogram",
+			nhcb: &Histogram{
+				CustomValues:    []float64{1, 2, 3},
+				PositiveBuckets: []int64{10, 20, 30}, // Delta format: {10, 20, 30} -> Absolute: {10, 30, 60}
+				Count:           60,
+				Sum:             100.0,
+			},
+			labels: labels.FromStrings("__name__", "test_metric"),
+			expected: ExpectedHistogram{
+				buckets: []BucketExpectation{
+					{le: "1", val: 10},
+					{le: "2", val: 30},
+					{le: "3", val: 60},
+					{le: "+Inf", val: 60},
+				},
+				count: 60,
+				sum:   100,
+			},
+		},
+		{
+			name: "Valid FloatHistogram",
+			nhcb: &FloatHistogram{
+				CustomValues:    []float64{1, 2, 3},
+				PositiveBuckets: []float64{20.0, 40.0, 60.0},
+				Count:           60.0,
+				Sum:             100.0,
+			},
+			labels: labels.FromStrings("__name__", "test_metric"),
+			expected: ExpectedHistogram{
+				buckets: []BucketExpectation{
+					{le: "1", val: 20},
+					{le: "2", val: 40},
+					{le: "3", val: 60},
+					{le: "+Inf", val: 60},
+				},
+				count: 60,
+				sum:   100,
+			},
+		},
+		{
+			name: "Empty Histogram",
+			nhcb: &Histogram{
+				CustomValues:    []float64{},
+				PositiveBuckets: []int64{},
+				Count:           0,
+				Sum:             0.0,
+			},
+			labels: labels.FromStrings("__name__", "test_metric"),
+			expected: ExpectedHistogram{
+				buckets: []BucketExpectation{
+					{le: "+Inf", val: 0},
+				},
+				count: 0,
+				sum:   0,
+			},
+		},
+		{
+			name: "Missing __name__ label",
+			nhcb: &Histogram{
+				CustomValues:    []float64{1, 2, 3},
+				PositiveBuckets: []int64{10, 20, 30}, // Delta format: {10, 20, 30} -> Absolute: {10, 30, 60}
+				Count:           60,
+				Sum:             100.0,
+			},
+			labels:    labels.FromStrings("job", "test_job"),
+			expectErr: true,
+		},
+		{
+			name:      "Unsupported histogram type",
+			nhcb:      nil,
+			labels:    labels.FromStrings("__name__", "test_metric"),
+			expectErr: true,
+		},
+		{
+			name: "Histogram with zero bucket counts",
+			nhcb: &Histogram{
+				CustomValues:    []float64{1, 2, 3},
+				PositiveBuckets: []int64{0, 10, 0}, // Delta format: {0, 10, 0} -> Absolute: {0, 10, 10}
+				Count:           10,
+				Sum:             50.0,
+			},
+			labels: labels.FromStrings("__name__", "test_metric"),
+			expected: ExpectedHistogram{
+				buckets: []BucketExpectation{
+					{le: "1", val: 0},
+					{le: "2", val: 10},
+					{le: "3", val: 10},
+					{le: "+Inf", val: 10},
+				},
+				count: 10,
+				sum:   50,
+			},
+		},
+		{
+			name: "Mismatched bucket lengths",
+			nhcb: &Histogram{
+				CustomValues:    []float64{1, 2},
+				PositiveBuckets: []int64{10, 20, 30}, // Mismatched lengths: 2 vs 3
```

Comment:
The name of the test is not already clear about it?
```suggestion
				PositiveBuckets: []int64{10, 20, 30},
```